src/Http/CurlHandler.php

```diff
@@ -169,6 +169,10 @@ protected function execute(array $options = [])
 
             $headerSize = $session->getInfo(CURLINFO_HEADER_SIZE);
 
+            if (strlen($result) < $headerSize) {
+                throw new Exception\TransferException('Header size(' . $headerSize . ') does not match result: ');
```

Comment:
Hmm, I do not think we should add it to SDK. Who want to debug, can override the class.